Assets/Unity3DTiles/B3DMLoader.cs

```diff
@@ -67,35 +77,47 @@ public IEnumerator LoadStream(string relativeFilePath)
+                    UInt32 byteLength = br.ReadUInt32();
+                    if (byteLength == 0)
+                    {
+                        Debug.LogError("Unexpected zero length in b3dm file: " + relativeFilePath);
+                    }
+                    // The length of the Feature Table JSON section in bytes.
                     UInt32 featureTableLength = br.ReadUInt32();
                     if (featureTableLength == 0)
                     {
                         Debug.LogError("Unexpected zero length feature table in b3dm file: " + relativeFilePath);
                     }
+                    // The length of the Feature Table binary section in bytes.
                     UInt32 featureBinaryLength = br.ReadUInt32();
                     if (featureBinaryLength != 0)
                     {
                         Debug.LogError("Unexpected non-zero length feature binary in b3dm file: " + relativeFilePath);
                     }
-                    // Batch table length
-                    br.ReadUInt32();
-                    if (featureBinaryLength != 0)
-                    {
-                        Debug.LogError("Unexpected non-zero length batch table in b3dm file: " + relativeFilePath);
-                    }
+                    // The length of the Batch Table JSON section in bytes. Zero indicates there is no Batch Table.
+                    UInt32 batchTableLength = br.ReadUInt32();
+                    // The length of the Batch Table binary section in bytes.
+                    // If batchTableJSONByteLength is zero, this will also be zero.
                     UInt32 batchBinaryLength = br.ReadUInt32();
-                    if (batchBinaryLength != 0)
+                    if (batchTableLength == 0 && batchBinaryLength != 0)
                     {
                         Debug.LogError("Unexpected non-zero length batch binary in b3dm file: " + relativeFilePath);
                     }
                     string featureTableJson = new String(br.ReadChars((int)featureTableLength));
                     FeatureTable ft = JsonConvert.DeserializeObject<FeatureTable>(featureTableJson);
-                    if (ft.BATCH_LENGTH != 0)
+                    if (batchTableLength == 0)
+                    {
+                        if(ft.BATCH_LENGTH != 0)
+                        {
+                            Debug.LogError("Unexpected non-zero length feature table BATCH_LENGTH in b3dm file: " + relativeFilePath);
+                        }
+                    } else
                     {
-                        Debug.LogError("Unexpected non-zero length feature table BATCH_LENGTH in b3dm file: " + relativeFilePath);
+                        string batchTableJson = new string(br.ReadChars((int)batchTableLength));
+                        BatchTable bt = JsonConvert.DeserializeObject<BatchTable>(batchTableJson);
                     }
+                    // after this will be the binary glTF
```

Comment:
well what about the batch table binary body if `batchBinaryLength > 0`?  Wouldn't you at least need to skip it here?  Or preferably, properly parse the batch table JSON as well as the binary body and resolve any references from the JSON into the binary data.